packages/next/src/lib/metadata/resolve-metadata.ts

```diff
@@ -358,7 +363,7 @@ async function getDefinedMetadata(
 ): Promise<Metadata | MetadataResolver | null> {
   // Layer is a client component, we just skip it. It can't have metadata exported.
   // Return early to avoid accessing properties error for client references.
-  if (isClientReference(mod)) {
+  if (!hasMetadataExport(mod)) {
```

Comment:
Should we adjust the comment above since this is no longer checking for a client reference?

Reply:
Oh good catch, I skipped it as it's too dimmed 🙈